gui/probe.lua

```diff
@@ -0,0 +1,71 @@
+-- Shows the info given by `probe` in a friendly display
+
+local gui = require('gui')
+local guidm = require('gui.dwarfmode')
+local widgets = require('gui.widgets')
+local overlay = require('plugins.overlay')
+
+Probe = defclass(Probe, widgets.Window)
+Probe.ATTRS{
+    frame = {
+        w = 40,
+        h = 45,
+        r = 2,
+        t = 18,
+    },
+    resizable=true,
+    frame_title='Probe',
+}
+
+function Probe:init()
+    self:addviews{
+        widgets.ToggleHotkeyLabel{
+            view_id='lock',
+            frame={t=0, l=0},
+            key='CUSTOM_CTRL_F',
+            label='Lock on tile:',
+            initial_option=false,
+        },
+        widgets.Label{
```

Comment:
since you're allowing the window to be resized, a `widgets.WrappedLabel` might be more appropriate here.

Reply:
Now that I have done it I find it a bit harder to understand this way. The wrapping is perfect for the tiletypes section, but on the materials section it becomes difficult to parse and to someone who is not already used to probe it is impossible to discern what it is describing because there are a lot of colons on the material names/description.

Wondering if having text clearly going off screen wouldn't be better to hint to the player that he should extend the screen if he needs that bit of info, but I don't know if it is even common knowledge that you can move and resize DFHack screens, considering vanilla screens don't let you I don't think anyone even tries.

Reply:
if the text needs better formatting, you have a few options:
1. change the formatting of `probe` itself
2. have `probe` return the raw data in a structure via a Lua API call and you can format it yourself

Reply:
Thinking a bit about it, I believe it would be very clear where and how wrapping is happening if we just color the words before the first colon on each line(if present). I don't suppose we have any markup or anything like that for the current label text pen situation? If not I think we are going with 2.